bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -178,24 +178,42 @@ export class PlotCanvasView extends BokehView
     # Update any DataRange1ds here
     frame = @model.frame
     bounds = {}
+    log_bounds = {}
+
+    calculate_log_bounds = false
+    for r in _.values(frame.x_ranges).concat(_.values(frame.y_ranges))
+      if r instanceof DataRange1d
+        if r._mapper_type == "log"
+          calculate_log_bounds = true
+
     for k, v of @renderer_views
       bds = v.glyph?.bounds?()
       if bds?
         bounds[k] = bds
+      if calculate_log_bounds
+        log_bds = v.glyph?.log_bounds?()
+        if log_bds?
+          log_bounds[k] = log_bds
 
     follow_enabled = false
     has_bounds = false
 
     for xr in _.values(frame.x_ranges)
       if xr instanceof DataRange1d
-        xr.update(bounds, 0, @model.id)
+        if xr._mapper_type == "log"
+          xr.update(log_bounds, 0, @model.id)
+        else
+          xr.update(bounds, 0, @model.id)
```

Comment:
Better code would be:
```
use_bounds = if xr._mapper_type == "log" then log_bounds else bounds
xr.update(use_bounds, 0, @model.id)
```
This way you avoid repetition, because the only difference is in bounds. If other arguments differed as well, then the original code would make more sense.